app/modules/annotations/models.py

```diff
@@ -187,6 +185,97 @@ def get_asset_src(self):
             assset_src = self.asset.src
         return assset_src
 
+    def get_matching_set(self, query=None, load=True):
+        if not self.encounter_guid:
+            raise ValueError(f'{self} has no Encounter so cannot be matched against')
+        if not query or not isinstance(query, dict):
+            query = self.get_matching_set_default_query()
+        log.info(f'finding matching set for {self} using query {query}')
+        return self.elasticsearch(query, load=load)
+
+    def get_matching_set_default_query(self):
+        # n.b. default will not take any locationId or ownership into consideration
+        parts = {'filter': []}
+        enc, enc_edm, sight_edm = self.get_related_extended_data()
+
+        # must have a content_guid (i.e. wbia guid)
+        parts['filter'].append(
+            {
+                'exists': {
+                    'field': 'content_guid',
+                }
+            }
+        )
+
+        viewpoint_list = self.get_neighboring_viewpoints()
+        # TODO should we allow nulls?
+        if viewpoint_list:
+            viewpoint_data = []
+            for vp in viewpoint_list:
+                viewpoint_data.append({'term': {'viewpoint': vp}})
+            parts['filter'].append(
+                {
+                    'bool': {
+                        'minimum_should_match': 1,
+                        'should': viewpoint_data,
+                    }
+                }
+            )
+
+        # same, re: nulls
+        tx_guid = self.get_taxonomy_guid(sighting_fallback=True)
+        if tx_guid:
+            parts['filter'].append({'match': {'taxonomy_guid': str(tx_guid)}})
+
+        if self.encounter_guid:
+            parts['must_not'] = {'match': {'encounter_guid': str(self.encounter_guid)}}
+
+        return {'bool': parts}
+
+    # this is to allow for manipulation of a user-provided query prior to actually using it
+    #  e.g. we might want to _force_ criteria or remove certain filters, etc.
+    def resolve_matching_set_query(self, query):
+        return query
+
+    # this should be used with caution.  it grabs the related Encounter object, as well as
+    # the edm data related to both the encounter and sighting.  thus, it is expensive.  it also
+    # caches these things on the Annotation itself so has implications there.   it is primarily
+    # intended for successive calls to find edm-properties (e.g. Elasticsearch indexing schema)
+    # and may not be ideal for "general usage" cases
+    def get_related_extended_data(self):
+        if hasattr(self, '_enc'):
+            return self._enc, self._enc_edm, self._sight_edm
+        if not self.encounter_guid:
+            return None, None, None
+        from app.modules.encounters.models import Encounter
+        from flask import current_app
+
+        self._enc = Encounter.query.get(self.encounter_guid)
```

Comment:
Set this last. If any of the EDM gets fail, you will be left with this set but no EDM data